### ISSUE TITLE
feat(sphere): exp-map flows (EMP, EMSRE) on S^d via autodiff Jacobians

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/gauss_flows#53]
## Problem / Request

Implement `ExpMapPolyFlow` (EMP) and `ExpMapSumRadFlow` (EMSRE) on S^d as `flowjax.bijections.AbstractBijection` subclasses with `shape=(d+1,)`. Both use the **tangent-space exponential map** as the transform; the log-determinant is computed via JAX **autodiff**, with **analytical formulas** living in `_src/transforms/_sphere_utils.py` as test oracles.

## User Story

> As a user with directional / spherical data, I want to stack expressive sphere-preserving bijections (EMP for cheap quadratic flows, EMSRE for radial-basis flows) and train them by maximum likelihood, so that I can model multimodal directional densities without leaving the manifold.

## Motivation

These two flow families form the foundation of Rezende et al. 2020 (the same paper PR #50 implements for the torus). EMP is the simplest expressive sphere flow; EMSRE adds capacity via a sum of radial bumps. Both generalise straightforwardly to arbitrary S^d.

## Proposed API

```python
# src/gauss_flows/_src/transforms/sphere.py

class ExpMapPolyFlow(AbstractBijection):
    """One layer of an exponential-map flow with quadratic scalar field.

    F(x) = μᵀx + xᵀ A x,    constrained so ‖μ‖₁ + ‖A‖₁ ≤ 1.

    Transform: tangent-space exp map of ∇F at x (projected onto T_x S^d).

    Shape:
        Input/output: (d+1,) unit vectors. log_det is a scalar Array.
    """
    shape: tuple[int, ...]   # = (d + 1,)
    cond_shape: ClassVar[None] = None
    mu: Array     # (d+1,) — paramax-constrained
    A: Array      # (d+1, d+1) — paramax-constrained, jointly with mu

    def __init__(self, key: PRNGKeyArray, d: int): ...


class ExpMapSumRadFlow(AbstractBijection):
    """One layer of an exponential-map flow with radial-basis scalar field.

    F(x) = Σₖ αₖ · exp(βₖ · (μₖᵀx − 1)),
        with μₖ ∈ S^d,  βₖ > 0,  αₖ ≥ 0,  Σ αₖ ≤ 1.
    """
    shape: tuple[int, ...]   # = (d + 1,)
    cond_shape: ClassVar[None] = None
    mu: Array      # (K, d+1) — each row a unit vector via paramax
    beta: Array    # (K,) — positive via paramax (softplus)
    alpha: Array   # (K,) — simplex-constrained Σ ≤ 1 via paramax

    def __init__(self, key: PRNGKeyArray, d: int, n_components: int): ...


# Factory helpers, mirroring `coupling_gaussianization_flow`
def emp_flow(key, d: int, n_layers: int) -> Transformed: ...
def emsre_flow(key, d: int, n_layers: int, n_components: int) -> Transformed: ...
```

## Mathematical Notes

```text
Exp-map step (shared between EMP and EMSRE):

    g  = ∇F(x)                          Euclidean gradient
    v  = (I_{d+1} − x xᵀ) · g           project onto T_x S^d
    z  = x · cos(‖v‖) + (v / ‖v‖) · sin(‖v‖)   exp map on S^d

Log-determinant on the manifold:
    Let  J = dz/dx ∈ ℝ^{(d+1)×(d+1)}  and  E = tangent_basis(x) ∈ ℝ^{(d+1)×d}.
    Then  log|det dz/dx|_{T_x S^d → T_z S^d}  =  ½ · log det(EᵀJᵀJE).
    The (d+1)×(d+1) Jacobian is rank-deficient by one (along x); the basis
    projection extracts the d×d intrinsic part.

Implementation: use jax.jacfwd / jax.jacrev for J. Keep an analytical
formula for J in `_src/transforms/_sphere_utils.py` for use as a test
oracle (and as documentation of the closed form for advanced users).

EMP scalar field:
    F(x)    = μᵀx + xᵀAx
    ∇F(x)   = μ + (A + Aᵀ)x
    H_F(x)  = A + Aᵀ                    (used only by analytical-J oracle)

EMSRE scalar field:
    F(x)    = Σₖ αₖ exp(βₖ (μₖᵀx − 1))
    ∇F(x)   = Σₖ αₖ βₖ exp(βₖ (μₖᵀx − 1)) · μₖ
    H_F(x)  = Σₖ αₖ βₖ² exp(βₖ (μₖᵀx − 1)) · μₖ μₖᵀ
```

## Design Snapshot

```python
# Sketch of transform_and_log_det using autodiff for J:

def transform_and_log_det(self, x: Array, condition=None) -> tuple[Array, Array]:
    def _z(x_in):
        g = jax.grad(self._F)(x_in)
        v = g - jnp.dot(x_in, g) * x_in    # project onto T_x S^d
        v_norm = jnp.linalg.norm(v)
        # Small-angle safe combine
        cos_v = jnp.cos(v_norm)
        sinc_v = jnp.sinc(v_norm / jnp.pi)  # sin(v_norm) / v_norm, NaN-safe
        return x_in * cos_v + v * sinc_v

    z = _z(x)
    J = jax.jacrev(_z)(x)                   # (d+1, d+1)
    E = tangent_basis(x)                    # (d+1, d)
    # log|det| of d×d intrinsic Jacobian
    M = E.T @ J.T @ J @ E
    sign, logabsdet = jnp.linalg.slogdet(M)
    return z, 0.5 * logabsdet
```

## paramax-based constraints

```python
# EMP: ‖μ‖₁ + ‖A‖₁ ≤ 1 via joint L¹-normalisation
def _emp_resolve(raw_mu, raw_A):
    A_sym = (raw_A + raw_A.T) / 2          # symmetric A doesn't lose generality
    total = jnp.sum(jnp.abs(raw_mu)) + jnp.sum(jnp.abs(A_sym))
    scale = 1.0 / (1.0 + total)            # smooth squash to (0, 1]
    return scale * raw_mu, scale * A_sym

# EMSRE: μₖ ∈ S^d via row-wise L²-normalisation
mu = px.Parameterize(lambda raw: raw / jnp.linalg.norm(raw, axis=-1, keepdims=True),
                     jr.normal(key_mu, (K, d+1)))
# βₖ > 0 via softplus + min
beta = px.Parameterize(lambda raw: jnn.softplus(raw) + 1e-3, jnp.zeros(K))
# αₖ ≥ 0, Σ ≤ 1 via softmax-style constrained simplex
alpha = px.Parameterize(lambda raw: jnn.softmax(jnp.append(raw, 0.0))[:-1], jnp.zeros(K))
```

## Acceptance criteria

- [ ] `ExpMapPolyFlow` and `ExpMapSumRadFlow` instantiate for `d ∈ {1, 2, 3, 7}`
- [ ] Forward transform preserves the manifold: `‖z‖₂ == 1` to atol=1e-5 after the transform
- [ ] **Autodiff vs analytical Jacobian agreement**: `assert jnp.allclose(autodiff_logdet, analytical_logdet, atol=1e-5)` for both classes across `d ∈ {2, 3, 5}` and randomised parameters
- [ ] Inverse via `jax.lax.while_loop` (fixed-point iteration on `expmap_sphere(x, −v)`) **OR** documented as forward-only with a clear `NotImplementedError`. Pick one in the PR.
- [ ] Roundtrip if inverse is implemented: `assert jnp.allclose(inverse(transform(x)), x, atol=1e-4)`
- [ ] Density estimation against the 4-mode VMF benchmark from utils ([flows-on-sphere/utils.py:36-44](https://github.com/ryushinn/flows-on-sphere/blob/master/utils.py#L36-L44)): EMP with N=6 layers reaches KL ≤ 0.6; EMSRE with N=6, K=5 reaches KL ≤ 0.25. Lower than the paper would be a bonus.
- [ ] paramax-constrained parameters: NO in-place clamping anywhere in the file
- [ ] All bijections vmap/jit/grad-friendly, gradient flows through every parameter
- [ ] 95%+ line coverage on the new module
- [ ] Re-exported from `gauss_flows.__init__`

## Out of scope

- MS family — separate child issue (#C)
- SO(3) / S³ specific tricks — defer to follow-up
- Conditional flows (`cond_shape ≠ None`) — defer

## Estimate

~400 LOC implementation + ~250 LOC tests. Medium PR.

## Dependencies

Blocked by: #52 (`tangent_basis`, `UniformOnSphere`)
